index.js

```diff
@@ -1,71 +1,186 @@
-var app = require('app')
+var menubar = require('menubar')
 var BrowserWindow = require('browser-window')
+var app = require('app')
+
+var fs = require('fs')
+
+var os = require('os')
 var ipfsd = require('ipfsd-ctl')
-var multiaddr = require('multiaddr')
-var Tray = require('tray')
-var menu = require('menu')
-var argv = require('minimist')(process.argv.slice(1))
+var ipc = require('ipc')
+var open = require('open')
 
+var multiaddr = require('multiaddr')
 var WEBUIPATH = '/webui'
 
 var LOGO = __dirname + '/node_modules/ipfs-logo/ipfs-logo-256-ice.png'
 var TRAY_ICON = (os.platform() !== 'darwin' ? LOGO
                  : __dirname + '/node_modules/ipfs-logo/platform-icons/osx-menu-bar.png')
 
-// keep references around for gc purposes
-var mainWindow = null
-var mainTray = null
+// only place where app is used directly
+var IPFS_PATH_FILE = app.getDataPath() + '/ipfs-electron-app-node-path'
+
+var MENU_WIDTH = 240
 
-var wizard = function (err) {
-  mainWindow = new BrowserWindow({icon: LOGO, width: 800, height: 600})
-  mainWindow.loadUrl('file://' + __dirname + '/wizard/wizard.html')
+var mainWindow
+var Ipfs
+
+var error = function (err) {
+  mainWindow = new BrowserWindow({icon: LOGO,
+                                  'auto-hide-menu-bar': true,
+                                  width: 800,
+                                  height: 600})
+  mainWindow.loadUrl('file://' + __dirname + '/html/help.html')
   mainWindow.webContents.on('did-finish-load', function () {
+    console.log('in loaded')
     mainWindow.webContents.send('err', err.toString())
   })
 }
 
+var initialize = function (path, node) {
+  mainWindow = new BrowserWindow({icon: LOGO,
+                                  width: 800,
+                                  'auto-hide-menu-bar': true,
+                                  height: 600})
+  mainWindow.loadUrl('file://' + __dirname + '/html/initialize.html')
+  mainWindow.webContents.on('did-finish-load', function () {
+    ipc.emit('default-directory', path)
+  })
+
+  // wait for msg from frontend
+  ipc.on('initialize', function (opts) {
+    ipc.emit('initializing')
+    node.init(opts, function (err, res) {
+      if (err) {
+        ipc.emit('initialization-error', err + '')
+      } else {
+        ipc.emit('initialization-complete')
+        ipc.emit('node-status', 'stopped')
+        fs.writeFileSync(IPFS_PATH_FILE, path)
+      }
+    })
+  })
+}
+
+var startTray = function (node) {
+  var poll
+  var statsCache = {}
+
+  ipc.on('request-state', function (event) {
+    ipfsd.version(function (err, res) {
+      if (err) throw err
+      ipc.emit('version', res)
+    })
+
+    if (node.initialized) {
+      ipc.emit('node-status', 'stopped')
+    }
+  })
+
+  ipc.on('start-daemon', function () {
+    ipc.emit('node-status', 'starting')
+    node.startDaemon(function (err, ipfs) {
+      if (err) throw err
+      ipc.emit('node-status', 'running')
+      poll = setInterval(function () { pollStats(ipfs) }, 500)
```

Comment:
is this to refresh the menu? every 500ms may be intensive, particularly when minimized.

wonder if we can make the polling only happen while the menu is up?